### PR TITLE
Colorized compiler diagnostics if using Ninja

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

Adding colorized compiler diagnostics.

Screenshot 2021-10-29 at 12 00 43

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,11 +31,20 @@ endif()
 cmake_minimum_required(VERSION 3.15)
 project(RealmJS)
 
-# Use ccache if available
-find_program(CCACHE_PROGRAM ccache)
-if(CCACHE_PROGRAM)
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}")
-endif()
```
Comment (kraenhansen) [inline CMakeLists.txt:36]: Why would this not break the use of ccache?

Comment (kneth) [reply CMakeLists.txt:36]: It looks like a mistake.

File: CMakeLists.txt [hunk 2]
```diff
@@ -49,6 +66,13 @@ if(DEFINED CMAKE_JS_VERSION)
     include(NodeJSTargets)
 endif()
 
+if (NOT MSVC)
+    # Ninja buffers output so we need to tell the compiler to use colors even though stdout isn't a tty.
+    if("${CMAKE_GENERATOR}" STREQUAL "Ninja")
```
Comment (kraenhansen) [inline CMakeLists.txt:71]: Is there any reason this needs to be wrapped in a string template?
```suggestion
    if(CMAKE_GENERATOR STREQUAL "Ninja")
```

**Review by fronck — APPROVED** [review]
:+1:

**Pull request merged**